File: psychology_backend/src/main/java/com/psychology/services/ClientService.java
package com.psychology.services;

import java.time.LocalDate;
import java.time.LocalTime;
import java.util.List;

import com.psychology.dtos.CallDTO;
import com.psychology.dtos.RecordDTO;
import com.psychology.entities.Call;
import com.psychology.entities.Office;
import com.psychology.entities.Psychologist;
import com.psychology.entities.Record;
import com.psychology.exceptions.IncorrectDateTime;
import com.psychology.exceptions.NoAvailableOffice;
import com.psychology.exceptions.NotFoundException;

public interface ClientService {
	Iterable<Psychologist> getPsychologists();

	Iterable<Office> getOffices();
	
	List<LocalTime> getFreePsychologistTime(Integer psychologistId, LocalDate date) throws NotFoundException;
	
	List<Office> getFreeOffices(LocalDate date, LocalTime time, LocalTime duration) throws IncorrectDateTime;
	
	Record addRecord(RecordDTO recordDTO) throws NoAvailableOffice, IncorrectDateTime, NotFoundException;
	
	Call addCall(CallDTO callDTO);
}

File: psychology_frontend/src/API/ClientService.js
import axios from "axios";
const API_URL = "http://localhost:8080/api/";

export default class ClientService {
  static async getPsychologists() {
    const response = await axios.get(API_URL + "psychologists");
    return response;
  }

  static async getOffices() {
    const response = await axios.get(API_URL + "offices");
    return response;
  }

  static async getFreeTimes(id, date) {
    const response = await axios.get(API_URL + "times/" + id, {
      params: {
        date: date,
      },
    });
    return response;
  }

  static async getFreeOffices(date, time, duration) {
    const response = await axios.get(API_URL + "freeOffices", {
      params: {
        date: date,
        time: time,
        duration: duration,
      },
    });
    return response;
  }

  static async addRecord({ record }) {
    const response = await axios.post(API_URL + "addRecord", record);
    return response;
  }

  static async addCall({ call }) {
    const response = await axios.post(API_URL + "addCall", call);
    return response;
  }
}
